bundles/org.openhab.binding.openwebnet/src/main/java/org/openhab/binding/openwebnet/internal/discovery/OpenWebNetDeviceDiscoveryService.java

```diff
@@ -0,0 +1,164 @@
+import org.openhab.binding.openwebnet.OpenWebNetBindingConstants;
+import org.openhab.binding.openwebnet.handler.OpenWebNetBridgeHandler;
+import org.openwebnet4j.OpenDeviceType;
+import org.openwebnet4j.message.BaseOpenMessage;
+import org.openwebnet4j.message.Where;
+import org.openwebnet4j.message.WhereZigBee;
+import org.openwebnet4j.message.Who;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link OpenWebNetDeviceDiscoveryService} is responsible for discovering OpenWebNet devices connected to a
+ * bridge/gateway
+ *
+ * @author Massimo Valla - Initial contribution
+ */
+@NonNullByDefault
+public class OpenWebNetDeviceDiscoveryService extends AbstractDiscoveryService {
+
+    private static final Set<ThingTypeUID> SUPPORTED_THING_TYPES = OpenWebNetBindingConstants.DEVICE_SUPPORTED_THING_TYPES;
+
+    private static final int SEARCH_TIME = 60;
+
+    private final Logger logger = LoggerFactory.getLogger(OpenWebNetDeviceDiscoveryService.class);
+    private final OpenWebNetBridgeHandler bridgeHandler;
+    private final ThingUID bridgeUID;
+
+    public OpenWebNetDeviceDiscoveryService(OpenWebNetBridgeHandler handler) {
+        super(SEARCH_TIME);
+        bridgeHandler = handler;
+        bridgeUID = handler.getThing().getUID();
+        logger.debug("Constructor for bridge: '{}'", bridgeUID);
+    }
+
+    @Override
+    public Set<ThingTypeUID> getSupportedThingTypes() {
+        return OpenWebNetDeviceDiscoveryService.SUPPORTED_THING_TYPES;
+    }
+
+    @Override
+    protected void startScan() {
+        logger.info("------ startScan() - SEARCHING for DEVICES on bridge '{}' ({}) ...",
+                bridgeHandler.getThing().getLabel(), bridgeUID);
+        bridgeHandler.searchDevices();
+    }
+
+    @Override
+    protected void stopScan() {
+        logger.debug("------ stopScan() on bridge '{}'", bridgeUID);
+        bridgeHandler.scanStopped();
+    }
+
+    @Override
+    public void abortScan() {
+        logger.debug("------ abortScan() on bridge '{}'", bridgeUID);
+        bridgeHandler.scanStopped();
+    }
+
+    /**
+     * Create and notify to Inbox a new DiscoveryResult based on WHERE, OpenDeviceType and BaseOpenMessage
+     *
+     * @param where the discovered device's address (WHERE)
+     * @param deviceType {@link OpenDeviceType} of the discovered device
+     * @param message the OWN message received that identified the device (optional)
+     */
+    public void newDiscoveryResult(Where where, OpenDeviceType deviceType, @Nullable BaseOpenMessage baseMsg) {
+        logger.info("newDiscoveryResult() WHERE={}, deviceType={}", where, deviceType);
+        ThingTypeUID thingTypeUID = OpenWebNetBindingConstants.THING_TYPE_GENERIC_DEVICE; // generic device
+        String thingLabel = OpenWebNetBindingConstants.THING_LABEL_GENERIC_DEVICE;
+        Who deviceWho = Who.UNKNOWN;
+        switch (deviceType) {
+            case ZIGBEE_ON_OFF_SWITCH: {
```

Comment:
The brackets are not necessary.
```suggestion
            case ZIGBEE_ON_OFF_SWITCH:
```